lib/ethereum/base.rb

```diff
@@ -0,0 +1,21 @@
+require 'digest/sha3'
+require 'rlp'
+
+module Ethereum
+  module Base
+
+    BYTE_ZERO = "\x00".freeze
+    GTXCOST = 21000       # TX BASE GAS COST
+    GTXDATANONZERO = 68   # TX DATA NON ZERO BYTE GAS COST
+    GTXDATAZERO = 4       # TX DATA ZERO BYTE GAS COST
+    SECP256K1_N = 115792089237316195423570985008687907852837564279074904382605163141518161494337
+    UINT_MAX = 2**256 - 1
```

Comment:
We should extract these constants (common constants like `BYTE_ZERO`, gas costs with `GTX` prefix, secp256k1 specific ones) into different modules.

If ethereum-base doesn't include secp256k1 functions, I wonder if we can move `SECP256K1_N` to a better place?